src/index.js

```diff
@@ -6,6 +6,29 @@ export default function register(Component, tagName, propNames, options) {
 		inst._vdomComponent = Component;
 		inst._root =
 			options && options.shadow ? inst.attachShadow({ mode: 'open' }) : inst;
+
+		if (options && options.shadow && options.injectGlobalStyles) {
+			const defaults = {
+				target: document.head,
+				selector:
+					'style, link[rel="stylesheet"], link[rel="preload"][as="style"]',
+				filter: undefined,
+				observeOptions: { childList: true, subtree: true },
+			};
+
+			this.styleObserver = beginInjectingGlobalStyles(
+				inst.shadowRoot,
+				options.injectGlobalStyles === true
+					? defaults
+					: /* eslint-disable indent */
+					  {
```

Comment:
If we do follow through with the stylesheet approach mentioned earlier we can remove this block.